lib/email/receiver.rb

```diff
@@ -570,6 +570,14 @@ def parse_from_field(mail = nil)
 
       return unless mail[:from]
 
+      if mail[:reply_to] && mail[:reply_to].errors.blank?
+        mail[:reply_to].each do |address_field|
+          from_address = address_field.address
+          from_display_name = address_field.display_name.try(:to_s)
+          return [from_address&.downcase, from_display_name&.strip] if from_address["@"]
```

Comment:
If we make previous change, then you will not need &. as we will end with empty string in worse case

Reply:
And the same for from_address too. If this line is called, it is only because from_address contains an @, so it is definitely not empty.

Reply:
Just copied what we already had for `:from` underneath :) Will clean both paths up.

Reply:
Adding to the other comments: This line will result in an error if `from_address` is `nil`.

Reply:
> If this line is executed, it is only because from_address contains an @, so it is definitely not empty.

This is actually not the case; this test is used to catch a from address without an @, so I need to leave the check in

https://github.com/discourse/discourse/blob/fa984731abfa8713300b5a4ef19ad0d71fd46909/spec/components/email/receiver_spec.rb#L425-L427